### ISSUE TITLE
Feature Request: ISO Hash Check

### ISSUE THREAD

**Issue by coolrecep** [pbatard/rufus#2396]
I have seen a lot of situations where user downloaded a corrupt ISO, can’t boot Windows and blames Rufus for this. All of the official ISOs have hash values. There could be an option to check ISO hash before writing it to the USB drive.

**Comment by pbatard** [comment]
1. Rufus already detects truncated ISOs, which should account for a large part of corrupted downloads.
2. Corrupted downloads will usually produce very explicit write errors.
3. I have a very vague plan to do __possibly__ do that, eventually, but since Microsoft no longer provides random Windows ISOs SHA-1 (it only provides them if you do download an ISO, which excludes anything older than the latest Windows 10 and Windows 11), I'd have to rely on an external service, such as https://sha1.rg-adguard.net/, which, just as I have to do with Fido, may require some heavy investment to ensure that, when the makers of the website update the content, it continues to work.

In short, while I have some long term plan to look into something like you suggest, this is both very low priority and, despite what you might assume, not that straightforward to add (especially as it would require UI alteration, and right now, I'm not planning to alter anything from the main UI until I switch to redesign Rufus to use WinUI/XAML Islands, which is probably still __years away__), so I will close this request for now.
